deepchecks/vision/metrics_utils/semantic_segmentation_metrics.py

```diff
@@ -37,26 +48,78 @@ def update(self, output: Tuple[torch.Tensor, torch.Tensor]):
-                y_i_count = y_i.sum()
+            dice_per_class = (2 * tp_count_per_class + self.smooth) / \
+                             (gt_count_per_class + pred_count_per_class + self.smooth)
 
-                self._evals[class_id]['dice'] += (2*tp) / (y_pred_i_count + y_i_count)
+            for class_id in [int(x) for x in torch.unique(y[i])]:
+                self._evals[class_id]['dice'] += dice_per_class[class_id]
                 self._evals[class_id]['count'] += 1
 
     def compute(self):
         """Compute metric value."""
         sorted_classes = [int(class_id) for class_id in sorted(self._evals.keys())]
         max_class = max(sorted_classes)
-        scores_per_class = np.empty(max_class + 1)*np.nan
+        scores_per_class = np.empty(max_class + 1) * np.nan
 
         for class_id in sorted_classes:
             count = self._evals[class_id]['count']
             dice = self._evals[class_id]['dice']
             mean_dice = dice / count if count != 0 else 0
             scores_per_class[class_id] = mean_dice
         return scores_per_class
+
+
+class MeanIoU(Metric):
+    """Metric that calculates the mean IoU metric for each class.
+
+    See more: https://en.wikipedia.org/wiki/Jaccard_index
+        Parameters
+    ----------
+    threshold: float, default: 0.5
+        prediction value per pixel above which the pixel is considered True.
+    smooth: float, default: 1e-3
+        smoothing factor to prevent division by zero when the mask is empty.
+    """
+
+    def __init__(self, *args, threshold: float = 0.5, smooth=1e-3, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._evals = defaultdict(lambda: {'iou': 0, 'count': 0})
+        self.threshold = threshold
+        self.smooth = smooth
+
+    def reset(self) -> None:
+        """Reset metric state."""
+        super().reset()
+        self._evals = defaultdict(lambda: {'iou': 0, 'count': 0})
+
+    def update(self, output: Tuple[torch.Tensor, torch.Tensor]):
+        """Update metric with batch of samples."""
+        y_pred, y = output
+
+        for i in range(len(y)):
+            gt_onehot, pred_onehot = format_segmentation_masks(y[i], y_pred[i], self.threshold)
+            tp_count_per_class, gt_count_per_class, pred_count_per_class = segmentation_counts_per_class(
+                gt_onehot, pred_onehot)
+
+            iou_per_class = (tp_count_per_class + self.smooth) / \
+                            (gt_count_per_class + pred_count_per_class - tp_count_per_class + self.smooth)
+
+            for class_id in [int(x) for x in torch.unique(y[i])]:
+                self._evals[class_id]['iou'] += iou_per_class[class_id]
+                self._evals[class_id]['count'] += 1
+
+    def compute(self):
+        """Compute metric value."""
+        sorted_classes = [int(class_id) for class_id in sorted(self._evals.keys())]
+        max_class = max(sorted_classes)
+        scores_per_class = np.empty(max_class + 1) * np.nan
```

Comment:
why not -np.ones like in detections? and then if not class is found put -1 as score?

Reply:
Because then you got -1 appearing in the graphs... the metric_results_to_df function ignores np.nan and that's the wanted behavior